src/interfaces/interfaces.ts

```diff
@@ -46,7 +46,16 @@ namespace interfaces {
     prototype: T;
   }
 
-  export type ServiceIdentifier<T = unknown> = (string | symbol | Newable<T> | Abstract<T>);
+  export type IfAny<T, Y, N> = 0 extends (1 & T) ? Y : N;
+
+  export type PropertyServiceIdentifier = (string | symbol);
+  export type ServiceIdentifier<T = unknown> = (PropertyServiceIdentifier | Newable<T> | Abstract<T>);
+  export type BindingMap = Record<PropertyServiceIdentifier, any>;
+  export type BindingMapKey<T extends BindingMap> = keyof T & PropertyServiceIdentifier;
+  export type ContainerIdentifier<T extends BindingMap> = IfAny<T, ServiceIdentifier, BindingMapKey<T>>;
+  export type ContainerBinding<T extends BindingMap, K extends ContainerIdentifier<T> = any> = K extends keyof T ? T[K] :
```

Comment:
I know I'm super picky with names, sry about it. I understand this is the service type of a binding, but not a binding. Am I right?

Reply:
No worries. Naming stuff is the hardest part of the job! It's especially tricky because I'm not super well-versed in this library, so have probably just been naming stuff wrong.

This type `ContainerBinding<T, K>` is meant to represent the return value type of `container<T>.get(key: K): ContainerBinding<T, K>`. So I guess it's whatever you want to call that?

Reply:
Then I guess it's a `ServiceType` (`BindingMapServiceType` would be too long, wouldn't it be?)